zschema/keys.py

```diff
@@ -354,11 +354,20 @@ def set_defaults(cls, required=None, doc=None, category=None):
         cls.set_default("doc", doc)
 
 
+def get_key_path(path=_NO_ARG):
+    path = path or []
+    s = lambda x: isinstance(x, int) and ("[%d]" % x) or "." + x
```

Comment:
I'd do
```
s: lambda x: "[{}]".format(x) if isinstance(x, int) else x
return ".".join(s[v] for v in path)
```

Reply:
that works; I was trying to avoid the double-index e.g. `root.array.[0].leaf`